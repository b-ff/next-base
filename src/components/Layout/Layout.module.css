.header,
.main,
.footer {
  width: 100%;
}

.header,
.main {
  border-bottom: 1px solid #ddd;
}

.main {
  flex: 1;
  min-height: 300px;
}
