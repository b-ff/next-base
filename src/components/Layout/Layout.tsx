"use client";
import { FC, ReactNode } from "react";
import styles from "./Layout.module.css";
import { Header } from "../Header";
import { Footer } from "../Footer";
import { Flex } from "../Flex";
import Link from "next/link";

export type LayoutProps = {
  children: ReactNode;
};

/**
 * Main layout of the app
 */
export const Layout: FC<LayoutProps> = ({ children }) => {
  return (
    <>
      <Header />
      <main className={styles.main}>{children}</main>
      <Footer />
    </>
  );
};
