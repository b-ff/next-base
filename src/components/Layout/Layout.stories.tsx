import type { Meta, StoryObj } from "@storybook/react";

import { Layout } from "./Layout";

const meta: Meta<typeof Layout> = {
  component: Layout,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof Layout>;

export const Primary: Story = {
  args: {
    children: "Content",
  },
};
