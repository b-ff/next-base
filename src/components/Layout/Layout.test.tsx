import { Layout } from "./Layout";
import { render } from "@testing-library/react";

const mockChildren = "Content";

describe("Component Layout", () => {
  it("Should render as expected", () => {
    const { container } = render(<Layout>{mockChildren}</Layout>);
    expect(container).toMatchSnapshot();
  });
});
