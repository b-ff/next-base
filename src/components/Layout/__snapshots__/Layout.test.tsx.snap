// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Layout Should render as expected 1`] = `
<div>
  <header
    class="header"
  >
    <section
      style="display: flex; flex-direction: row; gap: var(--spacing_2x);"
    >
      <div>
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div>
         Â»
      </div>
    </section>
    <a
      href="#"
    >
      Sign In / Sign Up
    </a>
  </header>
  <main
    class="main"
  >
    Content
  </main>
  <header
    class="footer"
  >
    <span>
      <a
        href="https://github.com/b-ff"
        rel="noopener noreffer"
        target="_blank"
      >
        Slava Biryukov
      </a>
       
      x 
      2023
       
    </span>
  </header>
</div>
`;
